### PR TITLE
Build search page - issue 12

### LINKED ISSUE #12: Page: Build "search page"
It'd be nice to have a search page built for the fd. I don't think this would be MVP, but I think moving forward it's a must-have. Ideas for MVP user stories:

1. I can search for a type of service. For Example: "food" would return say, the food bank. We can use components from the categories page.

### PR DESCRIPTION
**Pull request by Alex-Cannon**
Close #12 

I built the search page. Below are steps to test the page:

Navigate to the search page
1. Type a query like "food" or "food bank"
2. Click "search"
3. It is expected to see a list of results with "food" in title, description, type, etc.
4. Click an org
5. You are redirected to /category/basic-needs/food/:id?backLink=/search?s=food
6. When you click the back arrow, you are redirected to /search?s=food INSTEAD of the category page.
The search page does this:

1. Pulls ALL sheets LOCALLY
2. Searches through relevant fields for the keywords provided by user. For example, when "food" is provided, returns orgs that have food in title, description, type, etc.
3. Display

### CODE REVIEW

File: src/pages/search/index.jsx [hunk 1]
```diff
@@ -1,5 +1,116 @@
...
+  myData.shift();
+  return myData.map(service => {
+    // Type, subtype, population, service provider, service description
+    let index = service + service[2] + service[3] + service[4] + service[5];
+    let id = service[1];
+    return { id, index };
+  });
+}
+
+const queryServices = (data, query) => {
```
Comment (Vpr99) [inline src/pages/search/index.jsx]: Curious why you chose not to use Fuse for the indexed search—I've had really great experiences with that before and I'd prefer to use it here if possible.

File: src/pages/service-detail/index.jsx [hunk 2]
```diff
@@ -55,7 +57,7 @@ const ServiceDetail = ({ match }) => {
   return (
     <Fragment>
       <TitleBar
-        backLink={`/categories/${categoryId}/${typeId}`}
+        backLink={new URLSearchParams(location.search).get("backLink") || `/categories/${categoryId}/${typeId}`}
```
Comment (Vpr99) [inline src/pages/service-detail/index.jsx]: This is great!

Comment (Alex-Cannon) [reply src/pages/service-detail/index.jsx]: Thanks!

File: src/pages/search/index.jsx (hunk 1 above)
Comment (Alex-Cannon) [reply src/pages/search/index.jsx]: I just looked at fuse and it looks pretty awesome. I'll implement it.

**Pull request merged**